### PR TITLE
[dvtv] Parse adaptive formats as well

### PR DESCRIPTION
**Pull request by jktjkt**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The old code hit an error when it attempted to parse the string ``adaptive`` for video height. Actually parsing the returned playlists is a good idea because it adds more output formats, including some audio-only-ones.

### CODE REVIEW

File: youtube_dl/extractor/dvtv.py [hunk 1]
```diff
@@ -79,13 +79,22 @@ def _parse_video_metadata(self, js, video_id):
...
+                    'url': video['file'],
+                    'ext': ext,
+                    'format_id': '%s-%s' % (ext, video['label']),
+                    'height': int(video['label'].rstrip('p')),
+                    'fps': 25,
+                })
+            elif video['type'] == 'application/dash+xml':
+                adaptive_formats = self._extract_mpd_formats(video['file'],
+                                                             video_id)
+                formats.extend(adaptive_formats)
```
Comment (dstftw) [inline youtube_dl/extractor/dvtv.py]: Separate variable is superfluous.

File: youtube_dl/extractor/dvtv.py [hunk 2]
```diff
@@ -79,13 +79,22 @@ def _parse_video_metadata(self, js, video_id):
...
+                formats.append({
+                    'url': video['file'],
+                    'ext': ext,
+                    'format_id': '%s-%s' % (ext, video['label']),
+                    'height': int(video['label'].rstrip('p')),
+                    'fps': 25,
+                })
+            elif video['type'] == 'application/dash+xml':
+                adaptive_formats = self._extract_mpd_formats(video['file'],
+                                                             video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/dvtv.py]: Should be non `fatal`.

File: youtube_dl/extractor/dvtv.py [hunk 3]
```diff
@@ -79,13 +79,22 @@ def _parse_video_metadata(self, js, video_id):
...
+                formats.extend(adaptive_formats)
+            elif video['type'] == 'application/vnd.apple.mpegurl':
+                adaptive_formats = self._extract_m3u8_formats(video['file'],
+                                                              video_id)
+                formats.extend(adaptive_formats)
```
Comment (dstftw) [inline youtube_dl/extractor/dvtv.py]: Same.

File: youtube_dl/extractor/dvtv.py (hunk 2 above)
Comment (jktjkt) [inline youtube_dl/extractor/dvtv.py]: Done.

File: youtube_dl/extractor/dvtv.py (hunk 1 above)
Comment (jktjkt) [inline youtube_dl/extractor/dvtv.py]: Done.

File: youtube_dl/extractor/dvtv.py (hunk 3 above)
Comment (jktjkt) [inline youtube_dl/extractor/dvtv.py]: Done.

**Comment by jktjkt** [comment]
I've pushed 7e0dc0f3f40d89168f905d42bb548d8928687766 which fixes @dstftw 's comments.

**Comment by dstftw** [comment]
Add a test of what this fix actually fixes.

**Comment by jktjkt** [comment]
@dstftw , commit 2cc14cc now adds a recent video that used to result in an exception. I've checked that the added test triggers an exception in ``master``, and that this patch restores functionality.

**Pull request closed without merging**